chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -543,8 +559,6 @@ class AutoClonetypeException(message: String) extends ChiselException(message)
   * }}}
   */
 abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
-  override def className = "Bundle"
-
```

Comment:
`className` is used by `Printable` for hardware pretty printing. I think just using the default Aggregate `.getClass.getSimpleName` is better for obvious reasons. I can't recall my original thinking, but I think it had to do with making this pretty printing look like printing a Map in Scala (since that's sort of what Records/Bundles are), but I think giving the name of the Bundle class is generally better.

My only question is what about anonymous or inner bundles? You'll get some pretty weird names but perhaps that's still desirable.

Reply:
Since the intention of this is for human debugging, I think providing better naming on a best-effort basis makes sense. So anonymous Bundles will probably give some nasty name. Inner bundles should work fine, it's one of the test cases (though not intentionally).

Random thought: maybe we could regex match anonymous bundle names and replace it with a more friendly AnonymousBundle? Might be more trouble than it's worth though, so I'd prefer to wait for user feedback first.

Reply:
Yeah I think that's sensible, at least `MyModule$anon$1` tells you more than just `Bundle`!